Library/Homebrew/extend/fileutils.rb

```diff
@@ -6,24 +7,15 @@ module FileUtils
   # Create a temporary directory then yield. When the block returns,
   # recursively delete the temporary directory.
   def mktemp(prefix=name)
-    # I used /tmp rather than `mktemp -td` because that generates a directory
-    # name with exotic characters like + in it, and these break badly written
-    # scripts that don't escape strings before trying to regexp them :(
+    Dir.mktmpdir(prefix, HOMEBREW_TEMP) do |tmp|
```

Comment:
:( On the plus side, it's pure ruby, so backporting from 1.8.6 shouldn't be *too* bad if you go that route.